src/components/progress-indicator/bl-progress-indicator.ts

```diff
@@ -0,0 +1,47 @@
+import { CSSResultGroup, html, LitElement, TemplateResult } from 'lit';
+import { customElement, property } from 'lit/decorators.js';
+import style from './bl-progress-indicator.css';
+
+export type ProgressIndicatorSize = 'small' | 'medium' | 'large';
+
+/**
+ * @tag bl-progress-indicator
+ * @summary Baklava Progress Indicator component
+ */
+
+@customElement('bl-progress-indicator')
+export default class BlProgress extends LitElement {
+  static get styles(): CSSResultGroup {
+    return style;
+  }
+
+  @property({ type: String })
+  size: ProgressIndicatorSize = 'medium';
+
+  @property({ type: String })
+  max = '100';
+
+  @property({ type: String })
+  value = '0';
+
+  @property({ type: Boolean })
+  failed = false;
+
+  connectedCallback() {
+    super.connectedCallback();
+
+    setTimeout(() => {
+      this.style.setProperty('--bl-progress-indicator-max', this.max);
+      this.style.setProperty('--bl-progress-indicator-value', this.value);
+    });
```

Comment:
Those values should be set inside setters of max and value attributes. Because they can be changed over time.

Reply:
Can you try to read those values from attributes in CSS instead of setting in JS? https://developer.mozilla.org/en-US/docs/Web/CSS/attr If this works, it'll be easier, I think.

Reply:
I tried to use `attr` before but it doesn't work for me.

When I check in host:before i can see the values on the screen like this;
```
:host {
  --bl-progress-indicator-max: attr(max);
  --bl-progress-indicator-value: attr(value);
}

:host::before {
  content: var(--max) var(--value);
}
```

But when I tried accessing the `progress-indicator` class, it didn't work.

```
.progress-indicator::before {
  content: var(--value);
}
```

Reply:
You can define values on `.progress-indicator` element by reading them from aria attributes that you already write:

```css
.progress-indicator {
   --max: attr(aria-valuemax);
   --value: attr(aria-value);
  ...
}

.progress-indicator::before {
   width: calc(100% / var(--max) * var(--value));
   ...
}
```

With this, you don't need to listen and update css variables in JS anymore. Also setting private variable on `:host` element moved them to light dom (outside of the component). We already have kinda a problem about that and here we can start -maybe- a better approach to always set internal variables on wrapping element inside component.

Reply:
Apparently browser support for using `attr()` for the properties other than `content` is very bad: https://caniuse.com/css3-attr We can dismiss this for now.

But my first comment is still valid.